users/services.py

```diff
@@ -128,3 +131,19 @@ def send_email_change_confirmation_email(user: User, new_email: str):
             "reset_link": reset_link,
         },
     )
+
+
+def register_user_to_campaign(
+    user: User, campaign_key: str, campaign_data: dict, project: Project
+):
+
+    try:
+        UserCampaignRegistration.objects.create(
+            user=user, key=campaign_key, details=campaign_data
+        )
+        if project is not None:
+            ProjectUserPermission.objects.create(
+                user=user, project=project, permission=ObjectPermission.FORECASTER
+            )
```

Comment:
Do we have any checks whether user is allowed to join given project? Otherwise, you can specify id of a private project and enroll yourself into

Reply:
Could you please also check `signup_api_view` as well? It also has `add_to_project`

Reply:
I am using that too, but this is for users that already have an account and they need to register to a campaign